process/block/shardblock.go

```diff
@@ -233,7 +233,10 @@ func (sp *shardProcessor) ProcessBlock(
 		return err
 	}
 
-	sp.blockChainHook.SetCurrentHeader(header)
+	err = sp.blockChainHook.SetCurrentHeader(header)
```

Comment:
Move this after: 	

err = sp.verifyCrossShardMiniBlockDstMe(header)
	if err != nil {
		return err
	}
	
	Line 315.